web/src/stream_list_sort.ts

```diff
@@ -156,8 +176,21 @@ export function sort_groups(stream_ids: number[], search_term: string): StreamLi
         }
     }
 
+    const folder_sections_sorted = [...folder_sections.values()].sort((section_a, section_b) =>
+        util.strcmp(section_a.section_title, section_b.section_title),
+    );
+
     // This needs to have the same ordering as the order they're displayed in the sidebar.
-    const new_sections = [pinned_section, normal_section, dormant_section];
+    const new_sections = [pinned_section, ...folder_sections_sorted, normal_section, dormant_section];
+
+    // Don't call it "other channels" if there's nothing above it.
+    if (
+        folder_sections_sorted.length === 0 &&
+        pinned_section.streams.length === 0 &&
+        pinned_section.muted_streams.length === 0
+    ) {
+        normal_section.section_title = $t({defaultMessage: "CHANNELS"});
+    }
```

Comment:
Does this mean if we only have "muted channels" and "non-muted channels" that their names are "OTHER CHANNELS" and "MUTED CHANNELS"? That may deserve attention in #design.

Reply:
"MUTED CHANNELS" isn't a typical folder name